project/Formatting.scala

```diff
@@ -18,15 +18,8 @@ object Formatting {
     ScalariformKeys.preferences in MultiJvm := setPreferences(ScalariformKeys.preferences.value)
   )
 
-  lazy val docFormatSettings = Seq(
-    ScalariformKeys.preferences := setPreferences(ScalariformKeys.preferences.value, rewriteArrowSymbols = false),
-    ScalariformKeys.preferences in Compile := setPreferences(ScalariformKeys.preferences.value, rewriteArrowSymbols = false),
-    ScalariformKeys.preferences in Test := setPreferences(ScalariformKeys.preferences.value, rewriteArrowSymbols = false),
-    ScalariformKeys.preferences in MultiJvm := setPreferences(ScalariformKeys.preferences.value, rewriteArrowSymbols = false)
-  )
-
-  def setPreferences(preferences: IFormattingPreferences, rewriteArrowSymbols: Boolean = true) = preferences
-    .setPreference(RewriteArrowSymbols, rewriteArrowSymbols)
+  def setPreferences(preferences: IFormattingPreferences) = preferences
+    .setPreference(RewriteArrowSymbols, true)
```

Comment:
um, we'll see if there will be complaints. Was pretty much confusion about "what's that weird symbol" when people copied and especially when they had wrong char encoding. Anyway, we often include snippets from ordinary test sources so we probably are inconsistent anyway.

It would be an excellent IDE feature to rewrite such things when rendering instead.

Reply:
Yeah... in fact there are some fonts that use ligatures to achieve sort of that effect (e.g. https://www.hanselman.com/blog/MonospacedProgrammingFontsWithLigatures.aspx) but I've never ended up using one

Reply:
I believe emacs does that... a pal had that for "lambda". Intellij can do that but it inserts the utf-8 char instead of just rendering it like that.

Anyway, let's see - should be OK to just have those

Reply:
On the same note, https://github.com/tonsky/FiraCode font seems to be getting traction right now and it supports ligatures. I have not tried it just because my GPU accelerated terminal does not support it yet. :)